draft-ietf-tls-dtls-connection-id.md

```diff
@@ -162,158 +161,64 @@ connection identifiers to other parties.
-with usage "cid_spare" as soon as possible. Note that an endpoint MAY ignore
-requests which it considers excessive (though they MUST be ACKed as usual).
+        ContentType type;
+        ProtocolVersion version;
+        uint16 epoch;
+        uint48 sequence_number;
+        opaque cid[cid_length];               // New field
+        uint16 length;
+        opaque fragment[DTLSCompressed.length];
+   } DTLSCompressed;
 
-
-# Record Layer Extensions
-
-This extension is applicable for use with DTLS 1.2 and DTLS 1.3. This extension
-can be used with the optimized DTLS 1.3 record layer format.
-
-{{dtls-record12}} and {{dtls-record13}} illustrate the record formats of DTLS 1.2
-and DTLS 1.3, respectively.
-
-~~~~
-  struct {
-     ContentType type;
-     ProtocolVersion version;
-     uint16 epoch;
-     uint48 sequence_number;
-     opaque cid[cid_length];               // New field
-     uint16 length;
-     select (CipherSpec.cipher_type) {
-        case block:  GenericBlockCipher;
-        case aead:   GenericAEADCipher;
-     } fragment;
-  } DTLSCiphertext;
+   struct {
+        ContentType type;
+        ProtocolVersion version;
+        uint16 epoch;
+        uint48 sequence_number;
+        opaque cid[cid_length];               // New field
+        uint16 length;
+        select (CipherSpec.cipher_type) {
+            case block:  GenericBlockCipher;
+            case aead:   GenericAEADCipher;
+        } fragment;
+   } DTLSCiphertext;
 ~~~~
 {: #dtls-record12 title="DTLS 1.2 Record Format with Connection ID"}
 
-~~~~
-  struct {
-     opaque content[DTLSPlaintext.length];
-     ContentType type;
-     uint8 zeros[length_of_padding];
-  } DTLSInnerPlaintext;
-
-  struct {
-     ContentType opaque_type = 23; /* application_data */
-     ProtocolVersion legacy_record_version = {254,253); // DTLSv1.2
-     uint16 epoch;                         // DTLS-related field
-     uint48 sequence_number;               // DTLS-related field
-     opaque cid[cid_length];               // New field
-     uint16 length;
-     opaque encrypted_record[length];
-  } DTLSCiphertext;
-~~~~
-{: #dtls-record13 title="DTLS 1.3 Record Format with Connection ID"}
-
-Besides the "cid" field, all other fields are defined in the DTLS 1.2 and
-DTLS 1.3 specifications.
-
 Note that for both record formats, it is not possible to parse the
-records without knowing if the connection ID is in use and how long
-it is.
-
-# Examples
+records without knowing how long the Connection ID is.
 
-Below is an example exchange for DTLS 1.3 using a single
-connection id in each direction.
+In order to help the receiver discerning a record bearing a Connection ID from
+one that doesn't, the ContentType of the record containing the cid field MUST
+be: application_data_with_cid(25) instead of application_data(23),
+alert_with_cid(26) instead of alert(24) and heartbeat_with_cid(27) instead of
+heartbeat(24).
```

Comment:
"I would write this somewhat differently. In order to allow a receiver to determine whether a record has CID or not, connections which have negotiated this extension use new record types for all protected records. The following table shows the record types to use."

Note you also need handshake. I wonder if anyone cares about heartbeat, though.

Reply:
ACK, and you're right: I forgot handshake.

(I'd add heartbeat for consistency.)